.storybook/helpers/horizontal-constraint-arg-type.ts

```diff
@@ -0,0 +1,25 @@
+import Constraints, { type TMaxProp } from '@commercetools-uikit/constraints';
+
+type ExtractNumericValues<T> = T extends infer U
+  ? U extends number
+    ? U
+    : never
+  : never;
+
+type TMinMax = ExtractNumericValues<TMaxProp>;
+
+type THorizontalConstraintArgType = {
+  defaultValue?: TMaxProp;
+  min?: TMinMax;
+  max?: TMinMax;
+};
+
+const horizontalConstraintArgType = (
+  { defaultValue, min, max }: THorizontalConstraintArgType = { defaultValue: 7 }
+) => ({
+  control: 'select',
+  options: Constraints.getAcceptedMaxPropValues(min, max),
+  defaultValue: defaultValue,
+});
```

Comment:
Setting the `defaultValue` like this doesn't seem to work
https://github.com/storybookjs/storybook/issues/21016

Reply:
But we don't need to provide the default value here, right.
To me this is more like the definition of the control and the default value should be passed as part of the args to the default story.
You can take a look at the `PrimaryButton` stories to see what I mean.

Reply:
Sure, no need to provide it here 👍. Just thought it would be a nice-to-have, considering that the API technically allows it. Additionally, for the `iconArgType` helper, making it a function with a default icon option could be a nice addition

Reply:
I get it, but I still think it would be beneficial to keep the default value out of the control definition.

I believe it's better to define the default values in the props of the default story so it's easier to reuse them in alternative variants (I saw this pattern in Storybook docs) or even in the args value section of the meta definition, as they serve that purpose.

Reply:
👍 Removed 4c97db5de73c40558da473f4d905c5c01c36bd64